src/nets/e2e_asr_attctc_th.py

```diff
@@ -740,13 +797,20 @@ def recognize_beam(self, h, recog_args, char_list):
                 for l in six.moves.range(1, self.dlayers):
                     z_list[l], c_list[l] = self.decoder[l](
                         z_list[l - 1], (hyp['z_prev'][l], hyp['c_prev'][l]))
-                hyp['c_prev'] = c_list
-                hyp['z_prev'] = z_list
-                hyp['a_prev'] = att_w
 
                 # get nbest local scores and their ids
-                local_scores = functional.log_softmax(self.output(z_list[-1]), dim=1).data
-                local_best_scores, local_best_ids = torch.topk(local_scores, beam, dim=1)
+                local_att_scores = functional.log_softmax(self.output(z_list[-1]), dim=1).data
+                if lpz is not None:
+                    local_best_scores, local_best_ids = torch.topk(
+                        local_att_scores, ctc_beam, dim=1)
+                    ctc_scores, ctc_states = ctc_prefix_score(hyp['yseq'], local_best_ids[0], hyp['ctc_state_prev'])
+                    local_scores = \
+                        (1.0 - ctc_weight) * local_att_scores[:, local_best_ids[0]] \
+                        + ctc_weight * torch.from_numpy(ctc_scores - hyp['ctc_score_prev'])
```

Comment:
@takaaki-hori if I understand it correctly, the paper you reference does not have `- hyp['ctc_score_prev']` element (line 11 of Algorithm 1). Could you please let me know whether there is indeed a difference in this implementation and whether it makes any difference?

Reply:
> @takaaki-hori if I understand it correctly, the paper you reference does not have `- hyp['ctc_score_prev']` element (line 11 of Algorithm 1). Could you please let me know whether there is indeed a difference in this implementation and whether it makes any difference?

Hi, did you got the answer? I would like to know the intuition behind `- hyp['ctc_score_prev']`. It looks like this line wants to reduce the scare of ctc_scores..